tests/integration/UnsafeSharedCacheTest.java

```diff
@@ -0,0 +1,51 @@
+package com.jvm.external;
+
+import java.io.IOException;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.net.URL;
+import java.net.URLClassLoader;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import org.junit.Test;
+
+import static org.hamcrest.core.StringContains.containsString;
+import static org.junit.Assert.assertThat;
+import static org.junit.Assert.assertTrue;
+
+public class UnsafeSharedCacheTest {
+
+  private static String OS = System.getProperty("os.name").toLowerCase();
+
+  @Test
+  public void test_jarsOnClassPath_areInTheSharedCache() throws IOException, URISyntaxException {
+    // No support for symlinks on Windows. Let's skip that.
+    if (OS.indexOf("win") == -1) {
```

Comment:
To reduce nesting a nice pattern is to check the opposite condition first and exit early. So in this case it would be something like
```
if (OS.indexOf("win") != -1) { System.exit(); } 
/* rest of code un-nested one level */
```
or something like that